src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepExecution.java

```diff
@@ -65,18 +51,23 @@ public boolean start() throws Exception {
             // TODO find some way of allowing ComputedFolders to hook into the listener code
             throw new AbortException("Waiting for non-job items is not supported");
         }
+
+        List<Action> actions = new ArrayList<>();
+        actions.add(new CauseAction(new Cause.UpstreamCause(invokingRun)));
+        actions.add(new BuildUpstreamNodeAction(node, invokingRun));
```

Comment:
Or I suppose you could simply subclass `UpstreamCause` to add the `FlowNode.id` information.

Reply:
It seems like this would make things more complicated elsewhere, without much benefit besides saving a few hundred bytes in cold storage.